### PR TITLE
Add unit tests for User presence bug

### PR DESCRIPTION
**Pull request by liamjm**
Further demonstrating #519.

Tests were added in #520, but this adds them directly to the unit test framework, rather than external tests that need the hardware.

To run just these tests, use
`cargo test  --features std,with_ctap1,verbose test_check_user_presence`

### CODE REVIEW

File: src/ctap/mod.rs [hunk 1]
```diff
@@ -3746,6 +3748,28 @@ mod test {
...
+        // This will always return timeout.
+        fn user_presence_timeout() -> UserPresenceResult {
+            Err(UserPresenceError::Timeout)
+        }
+
+        let mut env = TestEnv::new();
+        env.user_presence().set(user_presence_timeout);
+        let response = check_user_presence(&mut env, DUMMY_CHANNEL);
+        // TODO(https://github.com/google/OpenSK/issues/519): This should be timeout.
+        assert!(matches!(response, Ok(_)));
```
Comment (kaczmarczyck) [inline src/ctap/mod.rs:3770]: Feel free to add this commit to #521, then we can have the correct assert here right away no?

File: src/ctap/mod.rs [hunk 2]
```diff
@@ -69,6 +69,8 @@ use crate::api::connection::{HidConnection, SendOrRecvStatus};
 use crate::api::customization::Customization;
 use crate::api::firmware_protection::FirmwareProtection;
 use crate::api::upgrade_storage::UpgradeStorage;
+#[cfg(test)]
+use crate::api::user_presence::UserPresenceResult;
```
Comment (ia0) [inline src/ctap/mod.rs:73]: This could go at the beginning of the `mod test`.

File: src/ctap/mod.rs (hunk 1 above)
Comment (liamjm) [reply src/ctap/mod.rs:3770]: OK, done, added to #521.

File: src/ctap/mod.rs (hunk 2 above)
Comment (liamjm) [reply src/ctap/mod.rs:73]: Yes!

Done.

**Comment by kaczmarczyck** [comment]
This moved to #521, and all comments resolved there. Closing this PR.

**Pull request closed without merging**